src/parse.c

```diff
@@ -7501,7 +7505,12 @@ static tree_t p_component_declaration(void)
    pop_scope(nametab);
 
    consume(tEND);
-   consume(tCOMPONENT);
+   if (peek() != tCOMPONENT) {
+      require_std(STD_19, "optional end component");
+      optional(tCOMPONENT);
+   }
```

Comment:
The `optional(tCOMPONENT)` is redundant: we already know at this point the next token is not `tCOMPONENT`.

```suggestion
   if (peek() != tCOMPONENT)
      require_std(STD_19, "optional end component");
```